Documentation/tutorial/links_and_cross_references.md

```diff
@@ -0,0 +1,251 @@
+
+In Markdown link or auto link, you can add a hashtag (`#`) followed by the anchor name after UID. For example:
+
+```markdown
+<xref:uid_to_file#anchor_name>
+
+[link_text](xref:uid_to_file#anchor_name]
+```
+
+Both urls will resolve to `url_to_file#anchor_name` in output HTML.
+
+The link text still resolves to the title of the whole file. If it's not what you need, you can specify your own link text.
+
+> Hashtag in `xref` is always treated as separator between file name and anchor name. That means if you have `#` in UID, it has to be [encoded](https://en.wikipedia.org/wiki/Percent-encoding) to `%23`.
+
+Hash tag is not support in `@` form.
+
+## Cross reference between projects
+
+Another common need is to reference topics from an external project. For example, when you're writing the documentation for your own .NET library, you'll want to add some links that point to types in .NET base class library. DocFX gives you this functionality by exporting all UIDs in a project into a map file and you can import it in another project to use them.
+
+### Cross reference map file
+
+When building a DocFX project, there will be an `xrefmap.yml` generated under output folder. This file contains information for all topics that have UID defined and their corresponding urls. The format of `xrefmap.yml` looks like this:
+
+```yaml
+references:
+- uid: uid_of_topic
+  name: title_of_topic
+  href: url_of_topic.html
+  fullName: full_title_of_topic
+- ...
+```
+
+It's a YAML object that contains following properties:
+
+1. `references`: a list of topic information, each item contains following properties:
+  - `uid`: UID to a conceptual topic or API reference
+  - `name`: title of the topic
+  - `href`: url to the topic, which is an absolute url or relative path to current file (`xrefmap.yml`)
+  - `fullName`: doesn't apply to conceptual, means the full qualified name of API. For example, for String class, its name is `String` and full qualified name is `System.String`. This property is not used in link title resolve for now but reserved for future use.
+
+> Topic is not necessarily to be a file, it can also be a section inside a file. For example, a method in a class. In this case its url could be an anchor in a file.
+
+### Using cross reference map
+
+You can import a cross reference map file in your DocFX then all UIDs defined that file can be cross referenced.
+
+To use a cross reference map, add a `xrefMaps` config to `build` section of `docfx.json`:
+
+```json
+{
+  "build": {
+    "xrefMaps": [
+      "<path_to_xrefmap>"
+    ],
+    ...
+  }
+}
+```
+
+The value of `xrefMaps` could be a string or a list of strings that contain the path/url to cross reference maps.
+
+> DocFX supports reading cross reference map from a local file or a web location. It's recommended to deploy `xrefmap.yml` to the web site together with topic files so that others can directly use its url in `docfx.json` without need to download it to local.
```

Comment:
@vwxyzh I know this is not correct for now, need update after you finished your xref improvement work.